src/panels/developer-tools/state/developer-tools-state.ts

```diff
@@ -357,7 +364,7 @@ class HaPanelDevState extends LitElement {
       this._entity = undefined;
       this._state = "";
       this._stateAttributes = {};
-      return;
+      this._updateEditor();
```

Comment:
The return statement prevents the function from continuing and fetching state since the entityId is empty anyway. 
```suggestion
      this._updateEditor();
      return;
```

Reply:
🙈 Good catch !